zerver/views/realm.py

```diff
@@ -262,6 +265,20 @@ def update_realm(
             )
             data["notifications_stream_id"] = notifications_stream_id
 
+    if reportmsg_stream_id is not None:
+        if realm.reportmsg_stream is None or (realm.reportmsg_stream.id != reportmsg_stream_id):
+            new_reportmsg_stream = None
+            if reportmsg_stream_id >= 0:
+                try:
+                    new_reportmsg_stream = get_stream_by_id_in_realm(reportmsg_stream_id, realm)
+                except Stream.DoesNotExist as err:
+                    raise JsonableError(str(err))
+
+            do_set_realm_reportmsg_stream(
+                realm, new_reportmsg_stream, reportmsg_stream_id, acting_user=user_profile
+            )
+            data["reportmsg_stream_id"] = reportmsg_stream_id
```

Comment:
Let's move this block after `signup_notifications_stream_id`; it feels wrong to have something between those two.

Reply:
DONE